tv_apps/dashboard/test/unit/digital_clock_test.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+/* global DigitalClock, MockL10n */
+
+require('/shared/test/unit/mocks/mock_l10n.js');
+require('/js/digital_clock.js');
+
+suite('DigitalClock', function() {
+  var clock;
+  var digitalClock;
+  var startDate;
+  var realL10n;
+
+  function createElement(tagName, id) {
+    var element = document.createElement(tagName);
+    element.id = id;
+
+    return element;
+  }
+
+  suiteSetup(function() {
+    realL10n = window.navigator.mozL10n;
+    window.navigator.mozL10n = MockL10n;
+
+    document.body.appendChild(createElement('span', 'hours-tens-digit'));
+    document.body.appendChild(createElement('span', 'hours-units-digit'));
+    document.body.appendChild(createElement('span', 'minutes-tens-digit'));
+    document.body.appendChild(createElement('span', 'minutes-units-digit'));
+    document.body.appendChild(createElement('span', 'seconds-tens-digit'));
+    document.body.appendChild(createElement('span', 'seconds-units-digit'));
+    document.body.appendChild(createElement('span', 'today-is'));
+  });
+
+  suiteTeardown(function() {
+    var elements = document.getElementsByTagName('span');
+    for (var i=0; i < elements.length; i++) {
+      document.body.removeChild(elements[i]);
+    }
+    window.navigator.mozL10n = realL10n;
+  });
+
+  setup(function() {
+    startDate = new Date(2015, 4, 31);
+    clock = sinon.useFakeTimers(startDate.getTime());
+    digitalClock = new DigitalClock();
+    digitalClock.init();
+  });
+
+  teardown(function() {
+    digitalClock.stop();
+    clock.restore();
+  });
+
+  test('Should display current date and time once the clock is initialized',
+  function() {
+    clock.tick();
+    assert.equal(digitalClock.date.getTime(), startDate.getTime());
+    assert.equal(digitalClock.elements['hours-tens-digit'].textContent, '0');
```

Comment:
I would suggest not to use `digitalClock.elements['...']` since they are internal structure.